lib/storage/file.js

```diff
@@ -277,6 +277,28 @@ function File(bucket, name, options) {
 nodeutil.inherits(File, ServiceObject);
 
 /**
+ * Generate a padded aes-256 base64 key for encryption.
+ *
+ * @param {string} code - A user-friendly passphrase.
+ * @return {string} - The base64 aes-256 key.
+ */
+File.encrypt_ = function(code) {
+  var cipher = crypto.createCipher('aes-256-ctr', 'gcloud-node');
+
+  var paddingLength = 32 - code.length;
+  while (paddingLength--) {
+    code += ' ';
+  }
+
+  var key = cipher.update(code, 'utf8', 'base64') + cipher.final('base64');
+
+  return {
+    key: key,
+    hash: crypto.createHash('sha256').update(key, 'base64').digest('base64')
+  };
+};
```

Comment:
The whole File object will be "encrypted" when you call `encrypt`:

```js
var unencryptedFile = bucket.file('my-file')
var file = bucket.file('my-file').encrypt('passphrase') // encrypted

// encrypt an unencrypted file reference
unencryptedFile.encrypt('passphrase')
```

Here are some ways to encrypt as a one liner:

```js
// way #1
file.encrypt('passphrase').save('new data', function(err) {})

// way #2
fs.createReadStream('local-file.txt')
  .pipe(file.encrypt('passphrase').createWriteStream())

// way #3
var file = bucket.file('new-file').encrypt('passphrase')
bucket.upload('local-file.txt', { destination: file }, function(err) {})
```

If way #&shy;3 is too complex, we can add an option to `bucket.upload`:

```js
bucket.upload('local-file.txt', { encrypt: 'passphrase' }, function(err) {})
```